app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/model/MovieProvider.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.marcinwisniewski.intivetrainingapp.movielist.model
+
+import com.example.marcinwisniewski.intivetrainingapp.network.ApiFactory
+import com.example.marcinwisniewski.intivetrainingapp.utils.Constant
+import io.reactivex.Observable
+
+class MovieProvider {
+    companion object {
```

Comment:
This looks like java factory method.
Creating instances should be done via dependency injection.
Maybe put TODO here and create task for later improvement.

Reply:
Added TODO anotation - I will create a new ticket in jara for it and learn/ implement dependency injection for it.